junit-vintage-engine/src/test/java/org/junit/vintage/engine/discovery/VintageDiscoveryRequestResolverTests.java

```diff
@@ -58,19 +60,21 @@ void logsWarningWhenFilterExcludesClass() {
 			logRecord.getMessage());
 	}
 
-	public static class Foo {
```

Comment:
No. That should not be the case. A static nested class is perfectly valid as a test class in the eyes of JUnit 4. And a "member class" is non-static. 

But... if you believe otherwise, please submit a separate PR demonstrating the bug you think is lurking here.

In the interim, please revert your changes to the `Foo` and `Bar` classes.